prow/github/client_test.go

```diff
@@ -1083,3 +1085,62 @@ func TestListIssueEvents(t *testing.T) {
 		t.Errorf("Wrong event for index 1: %v", events[1])
 	}
 }
+
+func TestThrottle(t *testing.T) {
+	ts := simpleTestServer(
+		t,
+		"/repos/org/repo/issues/1/events",
+		[]ListedIssueEvent{
+			{Event: IssueActionOpened},
+			{Event: IssueActionClosed},
+		},
+	)
+	c := getClient(ts.URL)
+	c.Throttle(1, 2)
+	if c.client != &c.throttle {
+		t.Errorf("Bad client %v, expecting %v", c.client, &c.throttle)
+	}
+	if len(c.throttle.throttle) != 2 {
+		t.Fatalf("Expected two items in throttle channel, found %d", len(c.throttle.throttle))
+	}
+	if cap(c.throttle.throttle) != 2 {
+		t.Fatalf("Expected throttle channel capacity of two, found %d", cap(c.throttle.throttle))
+	}
+	events, err := c.ListIssueEvents("org", "repo", 1)
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	if len(events) != 2 {
+		t.Errorf("Expected two events, found %d: %v", len(events), events)
+	}
+	if len(c.throttle.throttle) != 1 {
+		t.Errorf("Expected one item in throttle channel, found %d", len(c.throttle.throttle))
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	go func() {
+		if _, err := c.ListIssueEvents("org", "repo", 1); err != nil {
+			t.Errorf("Unexpected error: %v", err)
+		}
+		if _, err := c.ListIssueEvents("org", "repo", 1); err != nil {
+			t.Errorf("Unexpected error: %v", err)
+		}
+		cancel()
+	}()
+	for ctx.Err() == nil {
+		select {
```

Comment:
I don't think the outer `select` statement is necessary, just the contents of the `default` case.

Reply:
Done